tswow-scripts/wotlk/std/CreatureOutfits/CreatureOutfits.ts

```diff
@@ -159,4 +165,109 @@ export class CreatureOutfit extends MainEntity<creature_template_outfitsRow> {
             offhand === 0 && !emptyWeaponOverride ? -1 : offhand);
         return this;
     }
+
+   private bakeModel() {
+        let modelID = 0
+        let combatReach = 1.5
+        let boundingRadius = 0.306
+
+        let extraInfo = DBC.CreatureDisplayInfoExtra.add(Ids.CreatureDisplayInfoExtra.id())
+            .DisplayRaceID.set(this.Race.get())
+            .DisplaySexID.set(this.Gender.get())
+            .SkinID.set(this.Skin.get())
+            .FaceID.set(this.Face.get())
+            .HairStyleID.set(this.Hair.get())
+            .HairColorID.set(this.HairColor.get())
+            .FacialHairID.set(this.FacialHair.get())
+            .BakeName.set(`${this.texture}.blp`)
+        
+        if(this.Head.get()) extraInfo.NPCItemDisplay.setIndex(0, std.Items.load(this.Head.get()).DisplayInfo.get())
```

Comment:
Could this bet setup in a loop iterating an array of slots, and using the index as it loops up to make this easier to understand? Something like:
let equipmentSlots = [this.Head, this.Shoulders, this.Shirt, this.Chest].......

for(let i=0;i<equipmentSlots.length;i++){
let slot = equipmentSlots[i]
if(slot != null)
  extraInfo.NPCItemDisplay.setIndex(i, std.Items.load(slot).DisplayInfo.get())
}